libs/openFrameworks/video/ofAVFoundationVideoPlayer.m

```diff
@@ -122,19 +122,33 @@ - (void)dealloc
 	// destroy player
 	if(self.player != nil) {
 		[_player removeObserver:self forKeyPath:kRateKey];
-		
 		self.player = nil;
 		[_player release];
 	}
-	
+
 #if USE_VIDEO_OUTPUT
 	if (self.videoOutput != nil) {
 		self.videoOutput = nil;
 	}
-	
 	dispatch_release(_myVideoOutputQueue);
+
 #endif
-	
+
+	if(self.assetReaderVideoTrackOutput != nil) {
+		self.assetReaderVideoTrackOutput = nil;
+	}
+
+	if(self.assetReader != nil) {
+		[self.assetReader cancelReading];
+		self.assetReader = nil;
+		self.assetReaderVideoTrackOutput = nil;
+		self.assetReaderAudioTrackOutput = nil;
+	}
+
+	if(self.asset != nil) {
+		self.asset = nil;
+	}
```

Comment:
is this really releasing the asset? - 
I think setting it to nil is not releasing it. 

ping @julapy @danoli3  @admsyn